File: Java/quick_sort.java
public class quick_sort {
    public static void sort(int arr[], int si, int ei){
        if(si>=ei){
            return;
        }
        int index = si-1;
        int pivot = arr[ei];
        for(int i=si; i<=ei;  i++){
            if(arr[i]<=pivot){
                index++;
                int temp = arr[i];
                arr[i] = arr[index];
                arr[index] = temp;
            }
        }
        printArr(arr);
        sort(arr, si, index-1);
        sort(arr, index+1, ei);
    }
    public static void printArr(int arr[]){
        for(int i=0; i<arr.length; i++){
            if(i==arr.length-1){
                System.out.println(arr[i]);
            } else {
                System.out.print(arr[i] + ", ");
            }
        }
    }
    public static void main(String args[]){
        int sample[] = {6, 5, 4, 1, 2, 3};
        sort(sample, 0, sample.length-1);
        printArr(sample);
    }
}

File: CPP/quickSort.cpp
#include <iostream>
using namespace std;

void quickSort(int arr[], int start, int end) {
    if(start >= end) {
        return;
    }
    int pivot = arr[end], ind = start, temp;
    for(int i = start; i <= end; i++) {
        if(arr[i] <= pivot) {
            temp = arr[ind];
            arr[ind] = arr[i];
            arr[i] = temp;
            ind++;
        }
    }
    ind--;
    quickSort(arr, start, ind-1);
    quickSort(arr, ind+1, end);
}

int main() {
    int arr[] = {6, 1 ,3 ,4, 2 ,4};
    quickSort(arr, 0, 5);

    for(int i = 0; i < 6; i++) {
        cout << arr[i] << "  ";
    }
    cout << endl;
}